htdocs/js/instrument_controlpanel_swaldeletedata.js

```diff
@@ -0,0 +1,19 @@
+function swalFunction(event) {
+  swal({
+    title: 'Please confirm deletion',
+    text: 'The instrument data will be deleted',
+    type: 'warning',
+    showCancelButton: true,
+    confirmButtonColor: '#3085d6',
+    cancelButtonColor: '#d33',
+    confirmButtonText: 'Yes, delete the data',
+  }, function(result) {
+    if (result) {
+      // swal does not work here. The window instantly disappears
+      alert('\t\t\t Deleted! (You will need to reload the page again)');
```

Comment:
The alert should be removed.

Reply:
yes but the data still appears. You need to reload another time for it to dissappear.

Reply:
@pierre-p-s I didn't notice that when testing.
Maybe have it as a note in the sweetalert dialogue for deleting. The alert function shouldn't be used. I'll also test again to see if there is something that can be done about the data.

Reply:
When I tried (Firefox), the page reloaded automatically with previous data. Manually reloading the page as ask in the modal window, cleared the old data.

Reply:
I'm getting that behaviour as well. I still don't think an alert function should be used in Loris the behaviour is kind of frowned upon in the design world.

Reply:
Ill get rid of the alert and add the text to the swal modal window

Reply:
@prioux hold on.

Try moving the following code in Loris/src/Middleware/UserPageDecorationMiddleware.php

```
// This should be moved out of the middleware and into the modules that need it,
        // but is currently required for backwards compatibility.
        // This should also come after the above call to handle() in order for updated data
        // on the controlPanel to be properly displayed.
        if (method_exists($page, 'getControlPanel')) {
            $tpl_data['control_panel'] = $page->getControlPanel();
        }
```

To line 154 and the refresh won't be necessary.

Reply:
@maltheism You tagged the wrong pierre - you want @pierre-p-s  not Pierre Rioux

Reply:
@johnsaigle oops, @pierre-p-s look at the code fix above for the refresh problem.

Reply:
@maltheism The fix you proposed did not work.

Reply:
@pierre-p-s did you remove the old code that was moved?

Reply:
for me when I add it to the .class.inc file and delete it from Loris/src/Middleware/UserPageDecorationMiddleware.php the control panel disappears completely.

Reply:
@pierre-p-s can you message me on slack because I would like to replicate.

Reply:
found the problem. Will be fixed tomorrow

Reply:
Added the fix. Can be tested and reviewed again